frontend/src/lib/k8s/api/v1/clusterApi.ts

```diff
@@ -92,22 +92,56 @@ export async function setCluster(clusterReq: ClusterRequest) {
   );
 }
 
-// @todo: needs documenting.
-
+/**
+ * deleteCluster sends call to backend remove a cluster from the config.
+ *
+ * Note: Currently, the use for the optional clusterID is only for the clusterID for non-dynamic clusters.
+ * It is not needed or used for dynamic clusters.
+ * @param cluster
+ * @param source
+ * @param clusterID
+ */
 export async function deleteCluster(
-  cluster: string
+  /** The name of the cluster to delete */
+  cluster: string,
+  /** The source of the cluster, either 'kubeconfig' or 'dynamic_cluster' */
+  source?: string,
+  /** Whether to remove the kubeconfig file associated with the cluster */
+  removeKubeConfig?: boolean,
+  /** The ID for a cluster, composed of the kubeconfig path and cluster name */
+  clusterID?: string,
+  // /** The origin of the cluster, e.g., kubeconfig path */
+  kubeconfigOrigin?: string,
+  // /** The original name of the cluster, used for kubeconfig clusters */
+  originalName?: string
 ): Promise<{ clusters: ConfigState['clusters'] }> {
+  let deleteURL;
+  const removeFromKubeConfig = removeKubeConfig ? 'true' : 'false';
+
+  // If the source is kubeconfig and the originalName is provided, and removeKubeConfig is true,
+  // we need to construct the URL differently to ensure the correct parameters are passed.
+  if (source === 'kubeconfig' && originalName && removeKubeConfig) {
+    // for non dynamic clusters, we need to use the original name as a query parameter to find the actual context in the kubeconfig
+    // and remove it from the kubeconfig file.
+    deleteURL = `/cluster/${cluster}?removeKubeConfig=${removeFromKubeConfig}&source=${source}&configPath=${kubeconfigOrigin}&originalName=${originalName}`;
```

Comment:
If we are sending the configPath, doesn't the source="kubeconfig" get redundant?
i.e. can we decide in the backend if we should do the same logic by just checking if configPath is sent in the request?
(honest question)